### PR TITLE
Open .xd file in Adobe XD app on double click in FileTree.

### PR DESCRIPTION
**Pull request by niteskum**
in this PR change is on Brackets Launch It will detect If XD is installed. If it is installed then a Dialog will be shown to user asking If user likes to open .xd file in Adobe XD as aexternal editor. If users says yes then  double click on .xd file in file tree will open file in Adobe XD App. 

@swmitra  @shubhsnov  @jha-g  @sobisht  @narayani28  Please review.

### CODE REVIEW

File: src/project/FileTreeView.js [hunk 1]
```diff
@@ -569,6 +571,10 @@ define(function (require, exports, module) {
                 if (this.state.clickTimer !== null) {
                     this.clearTimer();
                 }
+                if("xd" === FileUtils.getFileExtension(this.myPath()).toLowerCase() && PreferencesManager.getViewState("OpenXDFileInXDApp")) {
+                    brackets.app.OpenDocumentWithXDApp(this.myPath(), function () {});
```
Comment (shubhsnov) [inline src/project/FileTreeView.js]: See my comment on brackets-shell, you can make this callback optional by using _dummyCallback

File: src/project/FileTreeView.js [hunk 2]
```diff
@@ -39,7 +39,8 @@ define(function (require, exports, module) {
         LanguageManager   = require("language/LanguageManager"),
         FileTreeViewModel = require("project/FileTreeViewModel"),
         ViewUtils         = require("utils/ViewUtils"),
-        KeyEvent          = require("utils/KeyEvent");
+        KeyEvent          = require("utils/KeyEvent"),
```
Comment (shubhsnov) [inline src/project/FileTreeView.js]: Why is this included?

File: src/brackets.js [hunk 3]
```diff
@@ -371,6 +371,42 @@ define(function (require, exports, module) {
             });
         });
 
+        // Check If Adobe XD Application is installed on User Machine
```
Comment (shubhsnov) [inline src/brackets.js]: Should we really pollute `brackets.js` with this piece of code? Can we think about moving this somewhere else?
@swmitra for thoughts?

File: src/file/FileUtils.js [hunk 4]
```diff
@@ -526,6 +531,23 @@ define(function (require, exports, module) {
...
+     */
+    function canBracketsOpenFileWithGivenExtensions(ext) {
+        return !extListNotSupportedInBrackets.includes(ext);
+    }
+
+    /**
+     * @param {string} ext File Extensions to be added Not Supported List
+     *
+     */
+    function addExtensionToNotSupportedList(ext) {
```
Comment (shubhsnov) [inline src/file/FileUtils.js]: This feels a bit unnecessary since only `xd` will be added to the list and added functionality will never be used again.

File: src/brackets.js (hunk 3 above)
Comment (shubhsnov) [reply src/brackets.js]: Perhaps we can move all this to an extension without changing any code files?

File: src/file/FileUtils.js (hunk 4 above)
Comment (niteskum) [reply src/file/FileUtils.js]: Fixed
